Doc: split "General concepts" into several pages (#7232)
ocaml/dune: doc/concepts/ordered-set-language.rst

Ordered Set Language
====================

A few fields take an ordered set as argument and can be specified using a small
DSL.

> This DSL is interpreted by Dune into an ordered set of strings using the

Reviewer: ```suggestion
Dune interprets this DSL into an ordered set of strings using the
```

following rules:

- ``:standard`` denotes the standard value of the field when it's absent
- an atom not starting with a ``:`` is a singleton containing only this atom
- a list of sets is the concatenation of its inner sets
- ``(<sets1> \ <sets2>)`` is the set composed of elements of ``<sets1>`` that do
  not appear in ``<sets2>``

In addition, some fields support the inclusion of an external file using the
syntax ``(:include <filename>)``. For instance, this is useful when you need to
run a script to figure out some compilation flags. ``<filename>`` is expected to
contain a single S-expression and cannot contain ``(:include ...)`` forms.

> Note that inside an ordered set, the first element of a list cannot be

Reviewer: ```suggestion
Note that inside an ordered set, a list’s first element cannot be
```
